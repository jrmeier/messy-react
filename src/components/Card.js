import React, { useState } from "react";

export const Card = ({ product }) => {
  const [clicked, setClicked] = useState(false);
  const { name, image, type, brandName, price, storeName } = product

  return (
    <div style={{ border: "1px solid black" }} onClick={() => setClicked(!clicked)}>
      {clicked && (
        <div>
          <p>Price: {price}</p>
          <p>{storeName}</p>
        </div>
      )}
      <h1>{name}</h1>
      <img
        alt="alt"
        src={image}
        style={{ maxWidth: 100, maxHeight: "auto" }}
      ></img>
      <p>{type}</p>
      <p>{brandName}</p>
    </div>
  );
}

export default Card;
