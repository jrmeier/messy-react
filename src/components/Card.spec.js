import React from "react";
import { Card } from "./Card";


describe("Card", () => {
    it('should return a single card', () => {
        const product = {
            name: "World's Best Waffle Maker",
            image: "https://www.restaurantsupply.com/media/catalog/product/cache/acb79d03af3da2b97c59ded0fca57b36/w/a/waring-ww180x_004_1.jpg",
            type: "RETAIL",
            brandName: "Waring",
            price: "$1,000",
            storeName: "Wayfair"
        }
        expect(<Card product={product}/>).toMatchSnapshot();
    })
})