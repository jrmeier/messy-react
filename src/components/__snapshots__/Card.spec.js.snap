// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card should return a single card 1`] = `
<Card
  product={
    Object {
      "brandName": "Waring",
      "image": "https://www.restaurantsupply.com/media/catalog/product/cache/acb79d03af3da2b97c59ded0fca57b36/w/a/waring-ww180x_004_1.jpg",
      "name": "World's Best Waffle Maker",
      "price": "$1,000",
      "storeName": "Wayfair",
      "type": "RETAIL",
    }
  }
/>
`;
